src/app/portlets/dot-edit-page/content/components/dot-edit-page-workflows-actions/dot-edit-page-workflows-actions.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { MenuItem } from 'primeng/primeng';
+import { DotWorkflowActions } from '../../../../../shared/models/dot-workflow-actions/dot-workflow-actions.model';
+import { WorkflowService } from '../../../../../api/services/workflow/workflow.service';
+import { Observable } from 'rxjs/Observable';
+
+@Component({
+    selector: 'dot-edit-page-workflows-actions',
+    templateUrl: './dot-edit-page-workflows-actions.component.html'
+})
+export class DotEditPageWorkflowsActionsComponent implements OnInit {
+    @Input() inode: string;
+    @Input() label: string;
+
+    workflowsActions: Observable<MenuItem[]>;
+
+    constructor(private workflowsService: WorkflowService) {}
+
+    ngOnInit() {
+        this.workflowsActions = this.workflowsService
+            .getPageWorkflows(this.inode)
+            .map((workflows: DotWorkflowActions[]) => this.getWorkflowOptions(workflows));
```

Comment:
I think is better do it into the ngOnchange method because inode is a input

Reply:
Why? we always do it in the ```ngOnInit``` and as per Angular doc they do the same.